core/events/domain/event.py

```diff
@@ -358,36 +362,337 @@ def project(self, submission: Submission) -> Submission:
         return submission
 
 
-# TODO: consider representing some of these as distinct events/commands?
 @dataclass
-class UpdateMetadata(Event):
-    """Update the descriptive metadata for a submission."""
+class SetTitle(Event):
+    """Update the title of a submission."""
 
-    metadata: List[Tuple[str, Any]] = field(default_factory=list)
+    title: str = field(default='')
 
-    FIELDS = [
-        'title', 'abstract', 'doi', 'msc_class', 'acm_class',
-        'report_num', 'journal_ref', 'comments'
-    ]
+    MIN_LENGTH = 5
+    MAX_LENGTH = 240
+    ALLOWED_HTML = ["br", "sup", "sub", "hr", "em", "strong", "h"]
+
+    def __post_init__(self):
+        """Perform some light cleanup on the provided value."""
+        self.title = self._cleanup(self.title)
 
-    # TODO: implement more specific validation here.
     def validate(self, submission: Submission) -> None:
-        """The :prop:`.metadata` should be a list of tuples."""
+        """Validate the title value."""
         submission_is_not_finalized(self, submission)
-        try:
-            assert len(self.metadata) >= 1
-            assert type(self.metadata[0]) in [tuple, list]
-            for metadatum in self.metadata:
-                assert len(metadatum) == 2
-        except AssertionError as e:
-            raise InvalidEvent(self) from e
+        self._does_not_contain_html_escapes(submission)
+        self._acceptable_length(submission)
+        no_trailing_period(self, submission, self.title)
+        if self.title.isupper():
+            raise InvalidEvent(self, "Must not be all-caps")
+        self._check_for_html(submission)
+
+    def project(self, submission: Submission) -> Submission:
+        """Update the title on a :class:`.Submission`."""
+        submission.metadata.title = self.title
+        return submission
+
+    def _does_not_contain_html_escapes(self, submission: Submission) -> None:
+        """The title must not contain HTML escapes."""
+        if re.search(r"\&(?:[a-z]{3,4}|#x?[0-9a-f]{1,4})\;", self.title):
+            raise InvalidEvent(self, "Title may not contain HTML escapes")
+
+    def _acceptable_length(self, submission: Submission) -> None:
+        """Verify that the title is an acceptable length."""
+        N = len(self.title)
+        if N < self.MIN_LENGTH or N > self.MAX_LENGTH:
+            raise InvalidEvent(self, f"Must be between {self.MIN_LENGTH} and"
```

Comment:
I would be explicit about the title field when raising `InvalidEvent` (i.e. "Title must be between...")